### ISSUE TITLE
mpich-4.1 : mpich suite :Spawn tests hang when running through sbatch.

### ISSUE THREAD

**Issue by nikhilnanal** [pmodels/mpich#6804]
I am trying to run the tests in the test/mpi folder. I am using  the sbatch command to dispatch a script which calls the runtest script.
I am seeing that spawn tests hang when run using sbatch and I eventually get an application timeout. This does not happen when i run the runtests script directly without using sbatch. 

Here is my configure step for mpich build
./configure --prefix=<mpich install path> --with-libfabric=<Libfabric install_path> --disable-fortran --with-device=ch4:ofi  --without-ze

configure for test/mpi
./configure --with-mpi=<mpich install path> same as above 

Other tests work fine except spawn tests and thread/spawn error/spawn which give the following error with sbatch. here is an example of the error with spawn1.
Unexpected output in spawn1: [mpiexec@cb23.cluster] APPLICATION TIMED OUT, TIMEOUT = 180s
Program spawn1 exited without No Errors

what is the correct way to use mpich tests with sbatch?

**Comment by hzhou** [comment]
Can you run a spawn test program in sbatch directly?

**Comment by nikhilnanal** [comment]
No,  I tried running mpiexec -n1 ./spawn1 standalone via sbatch, but it still hangs. I see a bunch of sruns on the node running/trying to run hydra_pmi_proxy.

**Comment by hzhou** [comment]
So the issue is `sbatch` fixes the total number of processes you are able to launch via srun thus prevents new processes being spawned. Could search in Slurm's documentation and see if there is any configuration or options that can be tweaked?

**Comment by nikhilnanal** [comment]
I have tried a few combinations so far increasing ntasks ( max tasks run by sbatch) and cpus-per-task(> 1) options but they haven't worked. Is there a way to force(or configure) spawn tests use mpiexec instead of srun to run the hydra_pmi_proxy to see if that prevents the above scenario of conflict between sbatch and srun?

**Comment by hzhou** [comment]
You may try force using ssh with option `-launcher ssh`.

**Comment by nikhilnanal** [comment]
thanks setting HYDRA_LAUNCHER=fork worked.
